st2common/st2common/util/output_schema.py

```diff
@@ -51,6 +52,28 @@ def _validate_action(action_schema, result, output_key):
     schema.validate(final_result, action_schema, cls=schema.get_validator("custom"))
 
 
+def mask_secret_output(value, output_schema_result):
+    value = copy.deepcopy(value)
+    if (
+        value["action"]["runner_type"] != "local-shell-cmd"
+        and value["action"]["runner_type"] != "orquesta"
+        and value["action"]["runner_type"] != "announcement"
+        and value["action"]["runner_type"] != "inquirer"
+        and value["action"]["runner_type"] != "noop"
+        and value["action"]["runner_type"] != "remote-shell-script"
+        and value["action"]["runner_type"] != "winrm-cmd"
+        and value["action"]["runner_type"] != "winrm-ps-cmd"
+        and value["action"]["runner_type"] != "winrm-ps-script"
+    ):
+        output_key = value["runner"]["output_key"]
```

Comment:
orquesta runner does have output_key:
https://github.com/StackStorm/st2/blob/c15d9ba6f8e706522b5d96912ae09c74975cfc7f/contrib/runners/orquesta_runner/orquesta_runner/runner.yaml#L17

But you are right that `output_key` is not required:
https://github.com/StackStorm/st2/blob/c15d9ba6f8e706522b5d96912ae09c74975cfc7f/st2common/st2common/models/api/action.py#L114-L118
I wonder about (in a different PR) making it required so that output masking would work with these other runners too.

So, maybe do this instead:
```python
def mask_secret_output(value, output_schema_result):
    output_key = value["runner"].get("output_key")
    if not output_key:
        return output_schema_result
    value = copy.deepcopy(value)
    for key, spec in value["actions"]["output_schema"].items():
```

Also we don't need to use six anymore as we don't support python2. Just do `.items()`.